packages/flutter_tools/lib/src/asset.dart

```diff
@@ -60,6 +60,8 @@ class AssetBundle {
      int result = await _validateFlutterYamlManifest(manifest);
      if (result != 0)
        return result;
+    } else {
+      return 0;
```

Comment:
if there's a manifest, build() should do nothing?

this needs a comment explaining why, at a minimum. :-)